src/cpp/session/modules/SessionLinter.cpp

```diff
@@ -291,16 +300,35 @@ Error getAllAvailableRSymbols(const FilePath& filePath,
    Error error;
    if (module_context::isRScriptInPackageBuildTarget(filePath) ||
        filePath.isWithin(projects::projectContext().directory().childPath("tests")))
+   {
+      DEBUG("- Package file:'" << filePath.absolutePath() << "'");
       error = getAvailableSymbolsForPackage(filePath, documentId, pSymbols);
+   }
    else
+   {
+      DEBUG("- Project file:'" << filePath.absolutePath() << "'");
       error = getAvailableSymbolsForProject(filePath, documentId, pSymbols);
+   }
    
    // If the file is within the 'tests/testthat' directory, then assume
    // 'testthat' will be available for those tests.
    if (filePath.isWithin(projects::projectContext().directory().childPath("tests/testthat")))
    {
       PackageSymbolRegistry& registry = packageSymbolRegistry();
-      registry.fillNamespaceExports("testthat", pSymbols);
+      registry.fillNamespaceObjects("testthat", pSymbols, false);
+   }
+   
+   // If the file is named 'server.R', 'ui.R' or 'app.R', we'll implicitly
```

Comment:
Over-detecting here should be okay -- it just means that we'll go and (asynchronously) get information about the `shiny` package (if possible); if we later detect a `library(shiny)` call in that document then we'll automagically provide shiny completions.